src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.cc

```diff
@@ -654,6 +654,12 @@ StatusOr<uint32_t> OpenSSLFixSubversionNum(RawFptrManager* fptrManager,
 
 }  // namespace
 
+bool IsBoringSSL(const std::filesystem::path& openssl_lib) {
```

Comment:
For now this is simplistic and only covers the netty case since I haven't found a good way to generalize this. I haven't found any obvious symbols within a program linked with boringssl that can be relied on.

Reply:
It's also worth mentioning that BoringSSL [docs](https://github.com/google/boringssl/blob/0cc51a793eef6b5295b9e0de8aafb1d87a39e210/PORTING.md) explicitly state not to try and depend on specific symbols within a binary to test if it is in use.

> The OPENSSL_IS_BORINGSSL macro may also be used to distinguish OpenSSL from BoringSSL in configure scripts. Do not use the presence or absence of particular symbols to detect BoringSSL.

Reply:
Since the macro is not available to us, we may have to go against that guidance...unless we can find a better option.

Reply:
Yea, I agree. My goal was to avoid that unless it was absolutely necessary.